src/mantarray_desktop_app/sub_processes/data_analyzer.py

```diff
@@ -663,12 +654,16 @@ def _check_mag_analysis_statuses(self) -> None:
 
     def _start_recording_snapshot_analysis(self, recording_path: str) -> None:
         # TODO (9/16/22): this should be run in a thread so that this process is still responsive to main
-        # TODO (11/16/22): add error handling. If the analysis fails here, then snapshot_dfs[0] will raise a KeyError
-        snapshot_dfs = run_magnet_finding_alg({}, [recording_path], end_time=RECORDING_SNAPSHOT_DUR_SECS)
-        snapshot_dict = snapshot_dfs[0].to_dict()
-        snapshot_list = [list(snapshot_dict[key].values()) for key in snapshot_dict.keys()]
+        try:
+            snapshot_dfs = run_magnet_finding_alg({}, [recording_path], end_time=RECORDING_SNAPSHOT_DUR_SECS)
+            # KeyError gets raised when mag finding fails
+            snapshot_dict = snapshot_dfs[0].to_dict()
+            snapshot_list = [list(snapshot_dict[key].values()) for key in snapshot_dict.keys()]
+            mag_analysis_msg = {"time": snapshot_list[0], "force": snapshot_list[1:]}
+
+        except Exception:
```

Comment:
We should make sure this message is only displayed if the exception is the "unable to converge error", other wise let's display some general error msg like "something went wrong"